ports/rp2/modmachine.c

```diff
@@ -95,6 +95,12 @@ static void mp_machine_set_freq(size_t n_args, const mp_obj_t *args) {
     if (!set_sys_clock_khz(freq / 1000, false)) {
         mp_raise_ValueError(MP_ERROR_TEXT("cannot change frequency"));
     }
+    // Option: Set PERI clock to system clock again
+    // clock_configure(clk_peri,
+    //                 0,
+    //                 CLOCKS_CLK_PERI_CTRL_AUXSRC_VALUE_CLK_SYS,
+    //                 SYS_CLK_KHZ * KHZ,
+    //                 SYS_CLK_KHZ * KHZ);
```

Comment:
I'm not sure what this commented-out code is aiming to do.  When `set_sys_clock_khz()` is called, does that update the PERI clock or not?

Reply:
The whole change is about the inconsistency, that when machine.freq() is called to change the frequency, then PERI clock is set to 48 MHz as result of calling `set_sys_clock_khz()`. The change I suggested would set PERI clock always to 48MHz. Then, changing the CPU frequency would not affect e.g. the UART or PWM settings, at the penalty of limiting the peripherals baud rates.
 
This commented code is about the alternative suggestion to set PERI clock always to the CPU clock. Then there would be changes to the baud rates of the peripherals when changing the CPU clock, but at known rates.

Reply:
OK, I understand more.

So without this PR (ie current master), what is the value of PERI at start up, is it the system clock, or system clock divided by some factor?

Note that `machine.freq()` takes multiple arguments to specify the frequency of the buses (see eg stm32 port).  So maybe it's a good idea to allow the user to specify the PERI clock as the second (optional) argument to `machine.freq()`?  Something like:
```py
machine.freq(sys, peri=48000000)
```

Reply:
At the moment the PERI freq at boot is the system clock 125Mhz and it will be set to 48Mhz when calling machine.freq(n). 

> Note that machine.freq() takes multiple arguments to specify the frequency of the buses (see eg stm32 port).

Actually, the stm32 port lost this feature with commit 7d39db2.  machine.freq() takes just 1 argument.

Reply:
I changed machine.freq() to have a second boolean type agrument. If set to False or omitted, the peripheral frequency will be 48MHz, which is the default value now at boot. If set to True, the peripheral frequency is the same as the MCU. The clock section for the peripheral clock has no divider. The documentation is wrong. So the peripheral clock can only be one of the other clock sources, like USB clock or SYS clock. 
The API could also be changed to accept a frequency instead of a boolean value, with 48_000_000 and the value of the first argument as the permitted values. But then, an error check & message would have to be added.
As a required side change, machine.freq() now accepts up to 4 arguments. I made a separate PR for that 1 char change as well.

Reply:
> The API could also be changed to accept a frequency instead of a boolean value, with 48_000_000 and the value of the first argument as the permitted values. But then, an error check & message would have to be added.

It looks like this is how you ended up implementing it (ie not as a boolean).  I think what you did -- specifying a freq -- is the right way to do it.  That way is more general, and more consistent with stm32.

But do you think it's better to make the peri frequency a keyword-only argument?  That forces the user to be explicit and makes it even more future proof/general.  Eg `machine.freq(125000000, peri=48000000)`.

Reply:
> But do you think it's better to make the peri frequency a keyword-only argument? That forces the user to be explicit and makes it even more future proof/general. Eg machine.freq(125000000, peri=48000000).

The Python API is is extmod/modmachine.c. Making peri=xxx a keyword only argument would either require to change that or to move the API back to the rp2 port.

Reply:
> The Python API is is extmod/modmachine.c. Making peri=xxx a keyword only argument would either require to change that

Yes, if we want to make this a keyword-only argument, we'd need to change the API in `extmod/modmachine.c`.  We can do that if we think it's the best option.  All other ports can remain as they are (stm32 accepting 4 positional args, all other ports accepting 1 positional arg).

Reply:
OK, let's leave it as a positional argument.